content/posts/2021-01-08-gradle-custom-shortcut.md

```diff
@@ -0,0 +1,81 @@
+---
+title: "Stop wasting your Android gradle daemons"
+date: 2021-01-08T16:48:02+09:00
+draft: false
+---
+
+As you've probably noticed, the switch between your dear Android Studio and your terminal
+is not without some latency / weird error messages about some Gradle daemons that could not be reused.
+
+{{< image src="/assets/2021-01-08-gradle-custom-shortcut/not-reused.png" position="center" >}}
+
+The reason is because Android Studio has its own default JDK installation,
+which lets you use Android Studio without installing Java
+or be sure all of your team is using the same compiler.
+
+That's convenient for sure, but it's not something your Gradle project is aware of and so,
+if you're trying to use Gradle without Android Studio's help,
+you will probably fallback to your default system JDK,
+which is not the one that Android Studio is using.
+It can affect you in 2 ways:
+
+1. not re-using the existing Gradle daemon (!!)
+1. not re-using your build cache (if your default JDK has a different version than the AS's one)
```

Comment:
Here is an alternative text about why you should care about your Gradle daemons:

```suggestion

But why should you care? The [Gradle daemon](https://docs.gradle.org/current/userguide/gradle_daemon.html#sec:why_the_daemon) is one of the key feature in order to speed up your builds and so, not using it efficiently can damage your day to day productivity.
```

Reply:
the key feature -> the key features